src/ui/base-stats-overlay.ts

```diff
@@ -0,0 +1,124 @@
+import type { InfoToggle } from "../battle-scene";
+import { TextStyle, addTextObject } from "./text";
+import { addWindow } from "./ui-theme";
+import * as Utils from "../utils";
+import i18next from "i18next";
+import { globalScene } from "#app/global-scene";
+
+export interface BaseStatsOverlaySettings {
+    scale?:number; // scale the box? A scale of 0.5 is recommended
+    x?: number;
+    y?: number;
+    /** Default is always half the screen, regardless of scale */
+    width?: number;
+}
+
+const HEIGHT = 120;
+const BORDER = 8;
+const GLOBAL_SCALE = 6;
+const shortStats = [ "HP", "ATK", "DEF", "SPATK", "SPDEF", "SPD" ];
+
+export default class BaseStatsOverlay extends Phaser.GameObjects.Container implements InfoToggle {
+
+  public active: boolean = false;
+
+  private statsLabels: Phaser.GameObjects.Text[] = [];
+  private statsRectangles: Phaser.GameObjects.Rectangle[] = [];
+  private statsShadows: Phaser.GameObjects.Rectangle[] = [];
+  private statsTotalLabel: Phaser.GameObjects.Text;
+
+  private statsBg: Phaser.GameObjects.NineSlice;
+
+  private options: BaseStatsOverlaySettings;
+
+  public scale: number;
+  public width: number;
+
+  constructor(options?: BaseStatsOverlaySettings) {
+    super(globalScene, options?.x, options?.y);
+    this.scale = options?.scale || 1; // set up the scale
+    this.setScale(this.scale);
+    this.options = options || {};
```

Comment:
```suggestion
    this.options = options ?? {};
```
You should use `??` instead of `||` unless you explicitly want `false`/`0`/etc to be treated the same as `null`/`undefined`.